api/src/wfl/module/covid.clj

```diff
@@ -686,18 +686,14 @@
         (throw-on-method-config-version-mismatch methodConfigurationVersion)))
   executor)
 
-(defn ^:private import-snapshot!
-  "Return snapshot reference for `id` imported to `workspace` as `name`."
-  [{:keys [workspace] :as _executor}
-   {:keys [name id]   :as _snapshot}]
-  (rawls/create-snapshot-reference workspace id name))
-
 (defn ^:private from-source
   "Coerce `source-item` to form understood by `executor` via `fromSource`."
-  [{:keys [fromSource] :as executor}
-   source-item]
-  (cond (= "importSnapshot" fromSource) (import-snapshot! executor source-item)
-        :else (throw (ex-info "Unknown fromSource" {:executor executor}))))
+  [{:keys [workspace fromSource] :as executor}
+   {:keys [name id]              :as _source-item}]
+  (cond (= "importSnapshot" fromSource)
+        (rawls/create-or-get-snapshot-reference workspace id name)
+        :else
+        (throw (ex-info "Unknown fromSource" {:executor executor}))))
```

Comment:
`throw` from `when`.
```clojure
(defn ^:private from-source
  "Coerce `source-item` to form understood by `executor` via `fromSource`."
  [{:keys [workspace fromSource] :as executor}
   {:keys [name id]              :as _source-item}]
  (when-not (= "importSnapshot" fromSource)
    (throw (ex-info "Unknown fromSource" {:executor executor})))
  (rawls/create-or-get-snapshot-reference workspace id name))
```

Reply:
It's likely down the road that we may support multiple variants for `fromSource`, so the `cond` seemed appropriate for that future.